[TF] Bring batchNormalized in line with standard batchnorm formula

jekbradbury opened the pull request:
Most batchnorm implementations, including those in Keras and TF, use
vector scale and offsets treated elementwise, not scalars.

rxwei commented on stdlib/public/TensorFlow/Ops.swift:
```diff
@@ -1587,8 +1587,8 @@ public extension Tensor where Scalar : BinaryFloatingPoint {
     alongAxis axis: Int32,
-    offset: Scalar = 0,
-    scale: Scalar = 1,
+    offset: Tensor = Tensor<Scalar>(0),
```
I believe you can drop the `<Scalar>` signature.

jekbradbury committed:
[TF] Bring batchNormalized in line with standard batchnorm formula

Most batchnorm implementations, including those in Keras and TF, use
vector scale and offsets treated elementwise, not scalars.
